### PR TITLE
Expose more engine classes for the DLLs usage

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Internal: Exposed more engine classes for the DLLs usage. Prepared a migration guide for iteration 34.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: docs/updating/migration-to-34.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+	```js
+	// ❌ Old imports:
+	import RealTimeCollaborativeRevisionHistory from '@ckeditor/ckeditor5-real-time-collaboration/src/realtimecollaborativerevisionhistory';
+	// ✅ New imports:
+	import RealTimeCollaborativeRevisionHistory from '@ckeditor/ckeditor5-real-time-collaboration/src/realtimecollaborativerevisionhistory';
+	import RevisionHistory from '@ckeditor/ckeditor5-revision-history/src/revisionhistory';
+	```
+
+* **{@link module:real-time-collaboration/realtimecollaborativerevisionhistory~RealTimeCollaborativeRevisionHistory}** will also require {@link module:revision-history/revisionhistory~RevisionHistory}:
```
Comment (scofalik) [inline docs/updating/migration-to-34.md:38]: Can you remind me if there were any places where we needed to import and add `CloudServicesRevisionHistoryAdapter`?

Comment (scofalik) [reply docs/updating/migration-to-34.md:38]: Also, this sentence was copy pasted but the first part was not changed.

Comment (ma2ciek) [reply docs/updating/migration-to-34.md]: It's removed.

File: docs/updating/migration-to-34.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Important changes
+
+### Additional dependencies in CKEditor 5 Collaboration Features
+
+The {@link installation/advanced/dll-builds DLL builds} support was introduced for revision history. As a result, some imports, plugin requirements and cross-package dependencies have changed to allow for the new building process.
+
+From now on, additional plugins will be required, when the following CKEditor 5 collaboration features are added to the editor:
+
+* **{@link module:real-time-collaboration/realtimecollaborativerevisionhistory~RealTimeCollaborativeRevisionHistory}** will also require {@link module:revision-history/revisionhistory~RevisionHistory}:
```
Comment (scofalik) [inline docs/updating/migration-to-34.md:28]:
```suggestion
* **{@link module:real-time-collaboration/realtimecollaborativerevisionhistory~RealTimeCollaborativeRevisionHistory}** will also require adding {@link module:revision-history/revisionhistory~RevisionHistory} to the list of the editor plugins:
```

**Pull request merged**